targets/TARGET_NORDIC/TARGET_NRF5/sleep.c

```diff
@@ -75,6 +77,15 @@ void hal_sleep(void)
 
 void hal_deepsleep(void)
 {
+    if (us_ticker_is_enabled()) {
+        /* Disable timer used for us ticker. */
+        nrf_timer_task_trigger(NRF_TIMER1, NRF_TIMER_TASK_STOP);
```

Comment:
Disabling the ticker should update the variable accessed with `us_ticker_is_enabled`; why not add a function that takes care of that ? 
This state should also be stored so it can be used after wake up.